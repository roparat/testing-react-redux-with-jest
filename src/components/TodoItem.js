import React, { Component, PropTypes } from 'react'

import style from './TodoItem.css'

const TodoItem = ({ text, complete }) => (
  <li className={complete ? style.complete : ''}>{text}</li>
)

TodoItem.propTypes = {
  text: PropTypes.string,
  complete: PropTypes.bool,
}

TodoItem.defaultProps = {
  text: '',
  complete: false,
}

export default TodoItem
