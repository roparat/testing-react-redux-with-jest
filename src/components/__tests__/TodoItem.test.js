import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';
import TodoItem from '../TodoItem'

describe('TodoItem', () => {
  it('matches its snapshot - not complete', () => {
    const component = shallow(<TodoItem text="Not complete"></TodoItem>)

    expect(toJson(component)).toMatchSnapshot();
  })

  it('matches its snapshot - completed', () => {
    const component = shallow(<TodoItem text="Complete" complete></TodoItem>)

    expect(toJson(component)).toMatchSnapshot();
  });

  it('render correct structure', () => {
    const component = shallow(<TodoItem text={'Hello'} />)
    
    expect(component.is('li')).toBe(true)
  })

  it('has complete class if receiving complete prop', () => {
    const component = shallow(<TodoItem text="Hello" complete />);

    expect(component.hasClass('complete')).toBe(true)
    expect(component.text()).toBe('Hello')
  })
})
