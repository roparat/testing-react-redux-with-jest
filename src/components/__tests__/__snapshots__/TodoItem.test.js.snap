// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem matches its snapshot - completed 1`] = `
<li
  className="complete"
>
  Complete
</li>
`;

exports[`TodoItem matches its snapshot - not complete 1`] = `
<li
  className=""
>
  Not complete
</li>
`;
